src/ext/java/org/opentripplanner/ext/vectortiles/VectorTilesResource.java

```diff
@@ -89,15 +89,19 @@ public TileJson getTileJson(
       .filter(Predicate.not(Objects::isNull))
       .toList();
 
-    return new TileJson(
-      uri,
-      headers,
-      requestedLayers,
-      ignoreRouterId,
-      "vectorTiles",
-      envelope,
-      feedInfos
-    );
+    List<String> rLayers = Arrays.asList(requestedLayers.split(","));
+
+    var url = serverContext
```

Comment:
After further testing, this resource is never hit when entering a URL of the form `<host>/<basePath_override>/{layers}/tilejson.json`. Even with the vector layers in `router-config.json` at build and run time, OTP still expects an entry in the original form, e.g.  `otp/routers/default/vectorTiles/stops/tilejson.json` for this code to run.

Reply:
This is expected behaviour as the goal is not to change the path that OTP serves the tiles on but to change the paths in the `tilejson.json` response.

The changing of the path is done with a proxy and that's why you need this configuration:  so that the path in the tilejson.json matches the path that the proxy expects.

I think this is a sign that the documentation is not good enough and I'm going to be working on it.

Reply:
I've updated the documentation in 6a087d15e2baf48bac7038dba8d6c81eacc2e5e1

Does this explain how this should work?